components/locale/src/extensions/unicode/attribute.rs

```diff
@@ -0,0 +1,91 @@
+use std::ops::RangeInclusive;
+use std::str::FromStr;
+
+use crate::parser::errors::ParserError;
+use tinystr::TinyStr8;
+
+#[derive(Debug, PartialEq, Eq, Clone, Hash, PartialOrd, Ord, Copy)]
+pub struct Attribute(TinyStr8);
+
+const ATTR_LENGTH: RangeInclusive<usize> = 3..=8;
+
+impl Attribute {
+    /// A constructor which takes a utf8 slice, parses it and
+    /// produces a well-formed `Variant`.
```

Comment:
Again, do we want to use word Variant here? (since we do have variant handling in locale space)